benchmark/parse_output.py

```diff
@@ -0,0 +1,163 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, The WGS-HGT Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+#
+# Parse output files of HGT tools.
+#
+
+import click
+import sys
+
+
+def parse_trex(input_f):
+    """ Parse output of T-REX version 3.6.
+
+    Parameters
+    ----------
+    input_f: string
+        file descriptor for T-REX output results
+
+    Returns
+    -------
+    number_of_hgts: string
+        number of HGTs reported by a tool, or NaN if an error occurred
+    """
+    string = 'hgt : number of HGT(s) found = '
+    for line in input_f:
+        if string in line:
+            return line.split(string)[1].strip()
+    return "NaN"
+
+
+def parse_rangerdtl(input_f):
+    """ Parse output of RANGER-DTL version 1.0.
+
+    Parameters
+    ----------
+    input_f: string
+        file descriptor for RANGER-DTL output results
+
+    Returns
+    -------
+    number_of_hgts: string
+        number of HGTs reported by a tool, or NaN if an error occurred
+    """
+    string = "The minimum reconciliation cost is: "
+    for line in input_f:
+        if string in line:
+            return line.split("Transfers: ")[1].split(",")[0]
+    return "NaN"
+
+
+def parse_riatahgt(input_f):
+    """ Parse output of RIATA-HGT version 3.5.6.
+
+    Parameters
+    ----------
+    input_f: string
+        file descriptor for RIATA-HGT output results
+
+    Returns
+    -------
+    number_of_hgts: string
+        number of HGTs reported by a tool, or NaN if an error occurred
+    """
+    string = "There are "
+    for line in input_f:
+        if string in line:
+            return line.split(string)[1].split(" component(s)")[0]
+    return "NaN"
+
+
+def parse_jane4(input_f):
+    """ Parse output of Jane 4.
+
+    Parameters
+    ----------
+    input_f: string
+        file descriptor for RIATA-HGT output results
+
+    Returns
+    -------
+    number_of_hgts: string
+        number of HGTs reported by a tool, or NaN if an error occurred
+    """
+    string = "Host Switch: "
```

Comment:
Same comments about constants and only returning first time it founds that line.